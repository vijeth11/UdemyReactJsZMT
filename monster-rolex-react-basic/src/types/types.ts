export type Monster = {
    id: string;
    name: string;
    email:string;
  }

export type CardProps = {
    monsters: Monster[]
};