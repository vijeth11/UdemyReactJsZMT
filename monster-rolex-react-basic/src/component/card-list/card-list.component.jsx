import React from "react";
import './card-list.styles.css';
import { Card } from '../card/card.component';

export const CardList = (props) => {
    console.log(props);
    //props contains all the inputs passed to the component 
    //props.children will contain the html elements defined between this component selectors whenever used (ex:app.js)
    //
    return (
    <div className='card-list'>{
        props.monsters.map(monster => {           
            return (<Card key={monster.id} monster = {monster}></Card>)
        })
      }</div>
    );
}

//Note:
/*
* Also keep in mind that React component names must be capitalized.
*/