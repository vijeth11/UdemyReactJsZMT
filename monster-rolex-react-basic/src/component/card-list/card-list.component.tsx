import './card-list.styles.css';
import { Card } from '../card/card.component';
import { CardProps, Monster } from '../../types/types';

export const CardList = (props:CardProps) => {
    console.log(props);
    //props contains all the inputs passed to the component 
    //props.children will contain the html elements defined between this component selectors whenever used (ex:app.js)
    //
    return (
    <div className='card-list'>{
        props.monsters.map((monster:Monster) => {           
            return (<Card key={monster.id} monster = {monster}></Card>)
        })
      }</div>
    );
}

//Note:
/*
* Also keep in mind that React component names must be capitalized.
*/