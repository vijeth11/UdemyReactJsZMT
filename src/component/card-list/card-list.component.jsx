import React from "react";
import './card-list.styles.css';
export const CardList = (props) => {
    console.log(props);
    //props constains all the inputs passed to the component 
    //props.children will contain the html elements defined between this component selectors whenever used (ex:app.js)
    //
    return (<div className='card-list'>{props.children}</div>);
}