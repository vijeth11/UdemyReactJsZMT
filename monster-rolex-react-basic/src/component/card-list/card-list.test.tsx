import {shallow } from 'enzyme';
import { Monster } from '../../types/types';
import { CardList } from './card-list.component';

let mockData:Monster[] = [
    {
    email:"test@test.com",
    id:"1",
    name:"test"
},
{
    email:"test2@test.com",
    id:"2",
    name:"test2"
}
]

it("expect to render a card component",() => {
    expect(shallow(<CardList monsters={mockData}/>)).toMatchSnapshot();
    });