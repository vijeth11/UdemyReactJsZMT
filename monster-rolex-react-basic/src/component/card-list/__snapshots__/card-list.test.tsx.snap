// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render a card component 1`] = `
<div
  className="card-list"
>
  <Card
    key="1"
    monster={
      Object {
        "email": "test@test.com",
        "id": "1",
        "name": "test",
      }
    }
  />
  <Card
    key="2"
    monster={
      Object {
        "email": "test2@test.com",
        "id": "2",
        "name": "test2",
      }
    }
  />
</div>
`;
